pkg/scheduler/plugins/capacity/capacity.go

```diff
@@ -654,7 +652,7 @@ func (cp *capacityPlugin) checkHierarchicalQueue(attr *queueAttr) error {
 		attr.guarantee = totalGuarantee
 		cp.totalGuarantee = totalGuarantee
 		attr.realCapability = cp.totalResource
-		attr.deserved = cp.totalResource
+		attr.deserved = totalDeserved
```

Comment:
I guess that you will be interested in https://github.com/volcano-sh/volcano/issues/3947. We can decouple of the queues' quota from physical resources by add a new mode and avoid affecting existing features. FYI: https://github.com/volcano-sh/volcano/issues/3929#issuecomment-2563227733